Flask login

zfrank7777 opened the pull request.

zfrank7777 committed:
Delete redundant codes and files

zfrank7777 committed:
Modify user api models, and add password hash

zfrank7777 committed:
Build flask-login mechanism.

Update last_seen on logged in user request.

zfrank7777 commented:
The only thing undone is that the response of an unauthorized access to login_required API.
For now the only login_required API is delete_user.

amjltc295 commented on src/surfing_penguin/routes.py:
```diff
@@ -120,51 +121,11 @@ def post(self):
+@api.marshal_with(api_return_message)
+def unauthorized():
+    print("entering")
+    return {'message': "Please Login First"}
```
Please add a \# TODO tag if it's not done yet.

amjltc295 commented on src/surfing_penguin/__init__.py:
```diff
@@ -11,6 +12,9 @@
+login_manager.init_app(surfing_penguin)
 
 api = Api(surfing_penguin)
 
```
Please add a \# NOQA in the end of this file

amjltc295 commented on src/surfing_penguin/routes.py:
```diff
@@ -1,68 +1,54 @@
 
 
+@surfing_penguin.before_request
+def before_request():
```
It would be better to add comments for this function (and other not self-explained ones).

amjltc295 reviewed (changes requested):
Overall looks good. Just some minor parts to be modified.

zfrank7777 committed:
Add some minor modifications.

Add comment on before_request.
Add #noqa on init import.
Add #TODO on handling unauthorized access.
